esrally/mechanic/java_resolver.py

```diff
@@ -21,7 +21,7 @@
 from esrally.utils import jvm
 
 
-def java_home(car_runtime_jdks, cfg):
+def java_home(car_runtime_jdks, allow_bundled_jvm, cfg):
```

Comment:
How about we refactor the signature a bit? Ideally we could also get rid of the `cfg` object here and pass the chosen runtime JDK directly. I propose the following signature:

```python
def java_home(car_runtime_jdks, specified_runtime_jdk, provides_bundled_jdk)
```

where 

* `specified_runtime_jdk` is derived from `cfg.opts("mechanic", "runtime.jdk")`
* `provides_bundled_jdk` determines whether this Elasticsearch version ships with a bundled JDK